src/pages/Profile/Profile.jsx

```diff
@@ -3,22 +3,43 @@ import { Navbar, Sidebar, FollowContainer, Post } from "../../components/index";
 import "./Profile.css";
 import { ProfileEditModal } from "../../components/index";
 import { useParams } from "react-router-dom";
-import { getUserById } from "../../features/users/usersSlice";
+import { getUserById, editUserDetails } from "../../features/users/usersSlice";
 import { useDispatch, useSelector } from "react-redux";
 
 function Profile() {
   const [showModal, setShowModal] = useState(false);
+  const [profileImg, setProfileImg] = useState("");
   const [userPosts, setUserPosts] = useState();
   const user = useSelector((state) => state.users.user);
   const posts = useSelector((state) => state.posts.posts);
   const dispatch = useDispatch();
   let { id } = useParams();
   const signInUser = localStorage.getItem("username");
+  const token = localStorage.getItem("token");
 
   useEffect(() => {
     dispatch(getUserById({ id }));
   }, [id, showModal]);
 
+  const imageHandler = (e) => {
+    const reader = new FileReader();
+    reader.onload = () => {
+      if (reader.readyState === 2) {
+        setProfileImg(reader.result);
+      }
+    };
+    reader.readAsDataURL(e.target.files[0]);
+  };
+
+  useEffect(() => {
+    if (profileImg) {
+      const userData = { ...user, avatar: profileImg };
+      console.log(userData, "1st");
```

Comment:
if this console statement is not required, then you can remove.

Reply:
Yahh sure